application/views/scripts/navigation/dashboard.phtml

```diff
@@ -1,11 +1,22 @@
+<?php
+
+use ipl\Html\HtmlString;
+
+?>
 <div class="controls">
     <?= $tabs ?>
 </div>
 <div class="content">
     <?php foreach ($navigation as $item): /** @var \Icinga\Web\Navigation\NavigationItem $item */?>
     <a class="dashboard-link" href="<?= $this->url($item->getUrl(), $item->getUrlParameters()) ?>"<?= $this->propertiesToString($item->getAttributes()) ?>>
         <div class="link-icon">
-            <?= $this->icon($item->getIcon() ?: 'forward', null, array('aria-hidden' => true)) ?>
+        <?php
+            if (gettype($item->getIcon()) == 'object' && get_class($item->getIcon()) == 'ipl\Web\Widget\Icon') {
+                echo (new HtmlString($item->getIcon()));
```

Comment:
You shouldn't need `HtmlString` here. Also expect the icon to be still a string, so it should be `new Icon($item->getIcon())` instead.

And please add the `aria-hidden` attribute just like it's done for the old icons.

Why a string? `configuration.php` files should be as lightweight as possible. Menu entries are such a common thing, it shouldn't be required to create instances of `Icon` explicitly.

Reply:
✅